dashboard/models.py

```diff
@@ -0,0 +1,150 @@
+        max_length=20,
+        choices=[("FWD", "Передний"), ("RWD", "Задний"), ("AWD", "Полный")],
+        help_text="Тип привода"
+    )
+    body_type = models.CharField(max_length=20, help_text="Кузов (седан, хэтчбек и т.д.)")
+    color = models.CharField(max_length=50, help_text="Цвет")
+    engine_type = models.CharField(
+        max_length=20,
+        choices=[("petrol", "Бензин"), ("diesel", "Дизель"), ("hybrid", "Гибрид"), ("ev", "Электро")],
+        help_text="Тип двигателя"
+    )
+    engine_volume = models.DecimalField(max_digits=3, decimal_places=1, help_text="Объем двигателя, л")
+    horsepower = models.PositiveSmallIntegerField(help_text="Мощность, л.с.")
+
+    # Пробег и история
+    mileage = models.PositiveIntegerField(help_text="Пробег, км")
+    owners_count = models.PositiveSmallIntegerField(help_text="Количество владельцев")
+    vin = models.CharField(max_length=17, blank=True, help_text="VIN номер", error_messages={
+            "unique": ("A car with that vin is already exists.")})
+
+    # Состояние и документы
+    is_new = models.BooleanField(default=False, help_text="Новый или с пробегом")
+    state = models.CharField(
+        max_length=20, 
+        choices=[("excellent", "Отличное"), ("good", "Хорошее"), ("needs_repair", "Требует ремонта")],
+        help_text="Состояние"
+    )
+    pts_type = models.CharField(
+        max_length=10, 
+        choices=[("original", "Оригинал"), ("duplicate", "Дубликат")],
+        help_text="ПТС"
+    )
+
+    # Цена и размещение
+    price = models.DecimalField(max_digits=10, decimal_places=2, help_text="Цена, руб.")
+    city = models.CharField(max_length=100, help_text="Город продажи")
+    address = models.CharField(max_length=200, blank=True, help_text="Адрес")
+
+    # Фото из CarImage
+
+    # Дополнительные сведения
+    features = models.TextField(blank=True, help_text="Дополнительные опции и комплектация")
+
+    # Владелец и дата публикации
+    seller = models.ForeignKey(CustomUser, on_delete=models.CASCADE)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        verbose_name = "Машина"
+        verbose_name_plural = "Машины"
+        constraints = (
+            models.UniqueConstraint(
+                fields=['vin'],
+                name='car_vin_unique'
+            ),
+            models.CheckConstraint(
+                check=models.Q(year__gte=1500),
+                name='car_year_gte_1500'
+            ),
+            models.CheckConstraint(
+                check=models.Q(updated_at__gte=F('created_at')),
+                name='check_created_date'
+            ),
+        )
+
+    def clean(self):
+        # Проверяем, что марка есть в базе моделей
+        if self.brand in Brand.objects.all():
+            # Проверяем, что модель входит в допустимый список для этой марки
+            if self.model not in CarModel.objects.filter(brand=self.brand):
```

Comment:
можно оптимизировать запрос
```suggestion
            if self.model_id not in CarModel.objects.filter(brand=self.brand).only("id"):
```

Reply:
Применил